apps/storybook/components/funke/IdCardPinScreen.stories.tsx

```diff
@@ -0,0 +1,243 @@
+import {
+  Button,
+  Heading,
+  type HeroIcon,
+  HeroIcons,
+  Image,
+  Page,
+  Paragraph,
+  ProgressBar,
+  Stack,
+  XStack,
+  YStack,
+} from '@package/ui'
+import type { Meta, StoryObj } from '@storybook/react'
+import React, { useRef, useState } from 'react'
+import { StyleSheet, type TextInput } from 'react-native'
+import { Circle, Input } from 'tamagui'
+import { LinearGradient } from 'tamagui/linear-gradient'
+
+import { useArgs } from '@storybook/addons'
+
+const germanIssuerImage = require('../../../funke/assets/german-issuer-image.png')
+
+interface IdCardPinScreenProps {
+  pinLength: number
+  onPinComplete: (pin: string) => void
+  state: 'enterPin' | 'loading' | 'complete'
+  onGoToWallet: () => void
+  userName?: string
+}
+
+const stateMapping = {
+  enterPin: { title: 'Enter passport pin', icon: 'locked' },
+  loading: { title: 'Setting up identity', icon: 'loading' },
+  complete: { title: 'You wallet is ready', icon: 'complete' },
+} as const
+
+const iconMapping = {
+  locked: <HeroIcons.LockClosed color="$white" />,
+  loading: <HeroIcons.ArrowPath color="$white" />,
+  complete: <HeroIcons.ShieldCheck color="$white" />,
+} as const
+
+interface IdCardProps {
+  icon: keyof typeof iconMapping
+  issuerImage: string
+  userName?: string
+}
+
+function IdCard({ icon, issuerImage, userName }: IdCardProps) {
```

Comment:
We're going to use this in more places so i'd extract this into it's own component.

Reply:
It is it's own component?

Reply:
A separate ID Card UI component that can be re-used across multiple screens, as it's now only in this Pin screen.

Reply:
Ah yes, will do the extracting out of storybook soon